src/app.rs

```diff
@@ -0,0 +1,51 @@
+use crate::contract_leaderboard::get_contracts;
+use crate::db::get_db_pool;
+use crate::state::AppState;
+use axum::{routing::get, Router};
+use sqlx::PgPool;
+use std::sync::Arc;
+
+pub async fn get_app() -> Router {
+    let db_pool = get_db_pool().await;
+    get_app_with_db_pool(db_pool)
+}
+
+fn get_app_with_db_pool(db_pool: PgPool) -> Router {
+    let shared_state = Arc::new(AppState { db_pool });
+    tracing_subscriber::fmt::init();
+    Router::new()
+        .route("/contracts", get(get_contracts))
+        .with_state(shared_state)
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::contract_leaderboard::ContractEntry;
+    use axum::{body::Body, http::Request};
+    use tower::ServiceExt;
+
+    #[tokio::test]
+    async fn test_get_contracts_against_dev_db() {
+        let app = get_app().await;
+        let response = app
+            .oneshot(
+                Request::builder()
+                    .uri("/contracts")
+                    .body(Body::empty())
+                    .unwrap(),
+            )
+            .await
+            .expect("Request Failed");
+        assert_eq!(response.status(), 200);
+        let body = hyper::body::to_bytes(response.into_body()).await.unwrap();
+        let contracts: Vec<ContractEntry> = serde_json::from_slice(&body).unwrap();
+        assert_eq!(contracts.len(), 100);
+
+        for contract in contracts {
+            assert!(contract.address.len() > 0);
+            assert_eq!(&contract.address[..2], "0x");
```

Comment:
Mmm, this too does not seem like a good test to me. If there is a contract in the DB anywhere that doesn't start with "0x" as its address, we have bigger problems we need to go solve.

Reply:
I agree. As mentioned below these tests are only meant as a starting point for the CI. They are just meant to establish that the general setup "works", i.e. the server connects tot he db and passes through the data correctly. (for example it does not mix up "address" and "base_fee" fields or something like that).

Please Lmk if you have ideas for more meaningful assertions that we should add here when testing against the dev db.